tls/extensions/s2n_cert_status.c

```diff
@@ -80,7 +81,14 @@ int s2n_cert_status_recv(struct s2n_connection *conn, struct s2n_stuffer *in)
         /* We only support OCSP */
         return S2N_SUCCESS;
     }
-    conn->status_type = S2N_STATUS_REQUEST_OCSP;
+
+    /* status_type keeps track of the OCSP status of a connection when a client requests
+     * OCSP stapling from a server. status_type should not be updated if the server requested
+     * OCSP stapling.
```

Comment:
Hmmm why not?

Reply:
This is related to the `s2n_connection_is_ocsp_stapled` API mentioned in the description. `conn->status_type` is used by this API to determine if the connection is OCSP stapled. If this flag were to be updated if the server requested OCSP stapling, it would change the meaning of this API:

> Returns
1 if OCSP response was sent (if connection is in S2N_SERVER mode) or received (if connection is in S2N_CLIENT mode) during handshake, otherwise it returns 0.

Reply:
"The status_type variable is only used when a client requests OCSP stapling from a server. A server can request OCSP stapling from a client but it is tracked elsewhere."

Maybe more clear than what is currently written. Otherwise I am just left with a lot of questions.